app/models/contact.rb

```diff
@@ -5,9 +5,23 @@ class Contact < ApplicationRecord
   VALID_PHONE_REGEX = /\A\d{3}\-\d{3}-\d{4}\z/
   VALID_EMAIL_REGEX = /\A[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}\z/
 
+  before_validation :normalize_names
+
   validates :first_name, presence: {message: "can't be blank"} 
   validates :last_name, presence: {message: "can't be blank"}
-  validates :first_name, uniqueness: { scope: [:last_name, :user_id], message: "and Last name already exist for this user" }
+  
+  validates :first_name, uniqueness: { 
+    scope: [:last_name, :user_id], 
+    case_sensitive: false,
+    message: "and Last name already exist for this user"
+}
```

Comment:
I'm so glad you caught this issue in the code and took the time to correct it. Excellent job adding more to this validation in the Contacts model. These changes are very beneficial in keeping our data clean and the application overall more user-friendly.